Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/Bookmarks/BookmarksViewController.swift

```diff
@@ -268,24 +270,36 @@ class BookmarksViewController: SiteTableViewController, ToolbarUrlActionsProtoco
     return true
   }
   
-  fileprivate func configureCell(_ cell: UITableViewCell, atIndexPath indexPath: IndexPath) {
-    
+  fileprivate func configureCell(_ cell: BookmarkTableViewCell, atIndexPath indexPath: IndexPath) {
     // Make sure Bookmark at index path exists,
     // `frc.object(at:)` crashes otherwise, doesn't fail safely with nil
     if let objectsCount = bookmarksFRC?.fetchedObjectsCount, indexPath.row >= objectsCount {
-        fatalError("Bookmarks FRC index out of bounds")
+        assertionFailure("Bookmarks FRC index out of bounds")
+        return
     }
     
     guard let item = bookmarksFRC?.object(at: indexPath) else { return }
     cell.tag = item.objectID
     
+    // See if the cell holds the same bookmark. If yes, we do not have to recreate its image view
+    // This makes scrolling through bookmarks better if there's many bookmarks with the same url
+    let domainOrFolderName = item.isFolder ? item.displayTitle : (item.domain?.url ?? item.url)
+    let shouldReuse = domainOrFolderName != cell.domainOrFolderName
+    
+    cell.domainOrFolderName = domainOrFolderName
+    
     func configCell(image: UIImage? = nil, icon: FaviconMO? = nil) {
       if !tableView.isEditing {
         cell.gestureRecognizers?.forEach { cell.removeGestureRecognizer($0) }
         let lp = UILongPressGestureRecognizer(target: self, action: #selector(longPressedCell(_:)))
         cell.addGestureRecognizer(lp)
       }
       
+        if !shouldReuse {
+            return
+        }
+        
+        cell.imageView?.cancelFaviconLoad()
       cell.backgroundColor = .clear
```

Comment:
nit: Some sort of mismatch in spacing here

Reply:
this class has a mix of 2 and 4 space format, once you approve this PR i will reformat it, don't want to add noise to the PR until then